Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/TaskViewModel.kt

```diff
@@ -30,22 +36,20 @@ class TaskViewModel : ViewModel() {
 
     fun clear() {
         _result.value = 0
-        _isComplete.value = false
         cancelled = false
         mustRestartApp = false
         onResultDismiss = null
-        _isRunning.value = false
+        state.value = State.NOT_STARTED
     }
 
     fun runTask() {
-        if (isRunning.value == true) return
-        _isRunning.value = true
+        if (state.value == State.RUNNING) return
```

Comment:
This refactoring makes a property of the existing code more easily visible: we can call `runTask` on a `TaskViewModel` that has been completed and not yet cleared. Is that WAI, or a bug?